srk/src/polynomialLattice.ml

```diff
@@ -153,12 +163,11 @@ let intersect t1 t2 =
     ; affine_lattice
     ; affine_context
     }
- *)
 
 let subset t1 t2 =
   let aff1 = affine_basis t1 in
   let ideal1 = ideal t1 in
-  let polys1 = List.append (Rewrite.generators ideal1) aff1 in
+  let polys1 = (Rewrite.generators ideal1) @ aff1 in
   List.for_all (fun p -> member p t2) polys1
```

Comment:
Not sufficient --  e.g., imagine taking t2 to be the lattice w/ affine basis polys1.  Line 171 succeeds, but t2 doesn't (necessarily) contain all of ideal1, just its generators.

Reply:
Got too trigger-happy :) Strengthened to require that the ideal generators of t1 lies in the ideal of t2; otherwise, multiplying a generator of t1 by a non-integer exits t2.